src/bloggy_frontend/src/pages/PostDetail.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { Container, Spinner, Alert, Row, Col, Image } from 'react-bootstrap';
+import { useFetchPostById } from '../hooks/useFetchPostById';
+import '../styles/pagestyle.css'
+
+const PostDetail = () => {
+  const { id } = useParams();
+  const postId = Number(id);
+
+  const { data: post, error, isLoading } = useFetchPostById(postId);
```

Comment:
isLoading is true only for the first fetch, if you want to show loading spinner for each fetch you should use isFetching